### ISSUE TITLE
OpenSSL upgrade from 1.1.1k -> 1.1.1l Intel 64bit Mac OS X El Captin 10.11.6

### ISSUE THREAD

**Issue by francisuk1989** [Homebrew/homebrew-core#84337]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
Upgrading openssl@1.1
1.1.1k -> 1.1.1l
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Upgrading openssl@1.1
1.1.1k -> 1.1.1l
Via Mac OS X El Captain on a Intel x64

### What happened (include all command output)?

Franciss-Mac-Pro:~ francis$ brew upgrade
Warning: You are using macOS 10.11.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

==> Upgrading 1 outdated package:
openssl@1.1 1.1.1k -> 1.1.1l
==> Upgrading openssl@1.1
  1.1.1k -> 1.1.1l 

```
==> Downloading https://www.openssl.org/source/openssl-1.1.1l.tar.gz
Already downloaded: /Users/francis/Library/Caches/Homebrew/downloads/b6ccc5a2a602c2af3480bbcf1656bd9844595974ba60501871ac12504508e818--openssl-1.1.1l.tar.gz
==> perl ./Configure --prefix=/usr/local/Cellar/openssl@1.1/1.1.1l --openssldir=/usr/local/etc/openssl@1.1 no-ssl3 no-ssl3-method no-zlib darwin64-x86_64-cc enable-ec_nistp_64_gcc_128
==> make
Last 15 lines from /Users/francis/Library/Logs/Homebrew/openssl@1.1/02.make:
clang  -I. -Iinclude -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/etc/openssl@1.1\"" -DENGINESDIR="\"/usr/local/Cellar/openssl@1.1/1.1.1l/lib/engines-1.1\"" -D_REENTRANT -DNDEBUG  -MMD -MF crypto/rc2/rc2_ecb.d.tmp -MT crypto/rc2/rc2_ecb.o -c -o crypto/rc2/rc2_ecb.o crypto/rc2/rc2_ecb.c
clang  -I. -Iinclude -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/etc/openssl@1.1\"" -DENGINESDIR="\"/usr/local/Cellar/openssl@1.1/1.1.1l/lib/engines-1.1\"" -D_REENTRANT -DNDEBUG  -MMD -MF crypto/rc2/rc2_skey.d.tmp -MT crypto/rc2/rc2_skey.o -c -o crypto/rc2/rc2_skey.o crypto/rc2/rc2_skey.c
clang  -I. -Iinclude -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/etc/openssl@1.1\"" -DENGINESDIR="\"/usr/local/Cellar/openssl@1.1/1.1.1l/lib/engines-1.1\"" -D_REENTRANT -DNDEBUG  -MMD -MF crypto/rc2/rc2cfb64.d.tmp -MT crypto/rc2/rc2cfb64.o -c -o crypto/rc2/rc2cfb64.o crypto/rc2/rc2cfb64.c
clang  -I. -Iinclude -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/etc/openssl@1.1\"" -DENGINESDIR="\"/usr/local/Cellar/openssl@1.1/1.1.1l/lib/engines-1.1\"" -D_REENTRANT -DNDEBUG  -MMD -MF crypto/rc2/rc2ofb64.d.tmp -MT crypto/rc2/rc2ofb64.o -c -o crypto/rc2/rc2ofb64.o crypto/rc2/rc2ofb64.c
In file included from crypto/rand/rand_unix.c:38:
/usr/include/CommonCrypto/CommonRandom.h:35:9: error: unknown type name 'CCCryptorStatus'
typedef CCCryptorStatus CCRNGStatus;
        ^
crypto/rand/rand_unix.c:385:47: error: use of undeclared identifier 'kCCSuccess'
    if (CCRandomGenerateBytes(buf, buflen) == kCCSuccess)
                                              ^
2 errors generated.
make[1]: *** [crypto/rand/rand_unix.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

### What did you expect to happen?

Hopefully someone has this issue either on Newer Intel Mac's or has a fix?

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew upgrade
```

**Comment by SMillerDev** [comment]
From your output 

> Warning: You are using macOS 10.11.
We (and Apple) do not provide support for this old version.
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
old version.

**Comment by francisuk1989** [comment]
Not a problem @SMillerDev i thought someone maybe had this error also on other newer Intel Mac's

**Comment by SMillerDev** [comment]
If this happened on newer systems we would have caught and fixed it in the CI runs.

**Comment by francisuk1989** [comment]
@SMillerDev Do you know how i can go back to OpenSSL 1.1.1k with brew?

something like brew install openssl@1.1/1.1.1k as this worked fine for me.

**Comment by francisuk1989** [comment]
It was an issue with OpenSSL 1.1.1l https://github.com/openssl/openssl/issues/16487

**Comment by lprimak** [comment]
@francisuk1989 Separate OpenSSL issue created for the test failure (see above)

**Comment by lprimak** [comment]
@SMillerDev This issue basically bricks any older mac that relies on Homebrew. This is why this isn't "run-of-the-mill" "I don't support this old HW/SW"-type issue

**Comment by Bo98** [comment]
I've pushed a fix for this. Let me know if you have any further issues.

**Comment by francisuk1989** [comment]
> I've pushed a fix for this. Let me know if you have any further issues.

I just seen this https://github.com/Homebrew/homebrew-core/commit/11d6f725bce47d5d133ca1018bceaf8b2342be2a

Thanks for the fix!

**Comment by lprimak** [comment]
@Bo98 Trying it now... Thank you!

**Comment by francisuk1989** [comment]
> @Bo98 Trying it now... Thank you!

Remember to
Brew update
brew upgrade
brew cleanup
brew install xxxx

**Comment by lprimak** [comment]
The compilation problem is now fixed - thanks, however, the test still failse:
https://github.com/openssl/openssl/issues/16517 is not fixed

**Comment by francisuk1989** [comment]
> The compilation problem is now fixed - thanks, however, the test still failse:
> [openssl/openssl#16517](https://github.com/openssl/openssl/issues/16517) is not fixed

I remove the line in brew edit and went fine.

**Comment by lprimak** [comment]
@francisuk1989 Please comment on https://github.com/openssl/openssl/issues/16517 so they see I am not the only one having this issue, maybe it will spur them to make a fix sooner, thanks!

**Comment by Bo98** [comment]
When upstream patch the issue, I will apply the patch to the formula.

**Comment by carlocab** [comment]
I believe the `make test` issue should be fixed now too as of ee3402c75ab3cddd398b9958eda705c8a99154f4.
